pkg/build/controller/build/defaults/defaults.go

```diff
@@ -187,29 +198,38 @@ func addDefaultEnvVar(build *buildapi.Build, v kapi.EnvVar) {
 }
 
 func addDefaultLabel(defaultLabel buildapi.ImageLabel, buildLabels *[]buildapi.ImageLabel) {
-	found := false
 	for _, lbl := range *buildLabels {
 		if lbl.Name == defaultLabel.Name {
-			found = true
+			return
 		}
 	}
-	if !found {
-		*buildLabels = append(*buildLabels, defaultLabel)
-	}
+	*buildLabels = append(*buildLabels, defaultLabel)
 }
 
-func addDefaultNodeSelector(k, v string, selectors map[string]string) bool {
+func addDefaultNodeSelector(k, v string, selectors map[string]string) {
 	if _, ok := selectors[k]; !ok {
 		selectors[k] = v
-		return true
 	}
-	return false
 }
 
-func addDefaultAnnotations(k, v string, annotations map[string]string) bool {
+func addDefaultAnnotation(k, v string, annotations map[string]string) {
 	if _, ok := annotations[k]; !ok {
 		annotations[k] = v
-		return true
 	}
-	return false
+}
+
+func addDefaultToleration(t kapi.Toleration, tolerations *[]v1.Toleration) {
+	tol := v1.Toleration{}
+
+	if err := kapiv1.Convert_core_Toleration_To_v1_Toleration(&t, &tol, nil); err != nil {
+		panic(err)
```

Comment:
you can't call panic, that would crash the api server.  util.HandleErr to log it, and ultimately you need to bubble this error up to ApplyDefaults so it can be returned from there.